src/ui/popup.js

```diff
@@ -435,30 +437,36 @@ export default class Popup extends Evented {
      * Adds a CSS class to the popup container element.
      *
      * @param {string} className Non-empty string with CSS class name to add to popup container
+     * @returns {Popup} `this`
      *
      * @example
      * let popup = new mapboxgl.Popup()
      * popup.addClassName('some-class')
      */
     addClassName(className: string) {
+        this._classList.add(className);
         if (this._container) {
             this._container.classList.add(className);
```

Comment:
What do you think about overwriting classes with something like `this._applyClassList()` which would look like:

```js
_applyClassList () {
  if (!this._container) return;
  this._container.className = [...this._classList.values()].join(' ');
}
```

My main concern is that we have two independent sources of truth which we _hope_ remain synchronized. But since we don't allow a user-supplied container anyway, so I don't _think_ there should be a case where this would prevent you from accomplishing some behavior you couldn't simply achieve by using the supplied API.

However, this could be a breaking change if someone was previously manipulating the container class directly.

Reply:
I don't think there's a _valid_ reason for anyone to manipulate the container class directly right now. (Still doesn't mean that no one's doing it.)

It wouldn't be quite as simple as the code you propose: the popup DOM element has additional default classes. So this:
```js
 const popup = new Popup({className: 'some classes'});
 ```
creates this element:
```html
 <div class="mapboxgl-popup mapboxgl-popup-anchor-bottom yellow"></div>
```

Reply:
Good point! Just wanted to be sure.

Reply:
I did find one weird edge case: if the user toggles one of the default classes:

`popup.toggleClassName('mapboxgl-popup');`

This removes the item from the container while adding it to the list.

I'm sure we could fix this, but I'm not sure what the correct behavior would be or if it's worth the lift and added complexity just to solve a very specific edge case.